From e859c6bccdfd7ab7ac0bddfb7c05b2044e0b78ff Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Tue, 16 Jan 2018 03:12:16 +0200
Subject: [PATCH 20/25] legacy: Label old UIO sysfs

< 3.10 paths are needed by rmt_storage on msm8974.
3.10 paths are needed by rfs_access and energyawarness on msm8992.

Change-Id: Ia54fa786a34916df0352c0ca5e8053edd1ee6234
---
 legacy-common/energyawareness.te |  1 +
 legacy-common/file_contexts      | 12 ++++++++++++
 legacy-common/rfs_access.te      |  2 ++
 3 files changed, 15 insertions(+)
 create mode 100644 legacy-common/energyawareness.te

diff --git a/legacy-common/energyawareness.te b/legacy-common/energyawareness.te
new file mode 100644
index 0000000..23c691a
--- /dev/null
+++ b/legacy-common/energyawareness.te
@@ -0,0 +1 @@
+r_dir_file(energyawareness, sysfs_uio_file)
diff --git a/legacy-common/file_contexts b/legacy-common/file_contexts
index ec55bdd..062d5fc 100644
--- a/legacy-common/file_contexts
+++ b/legacy-common/file_contexts
@@ -35,3 +35,15 @@
 
 # Camera
 /data/fdAlbum                                                       u:object_r:camera_data_file:s0
+
+# UIO
+/sys/devices(/platform)?/soc\.0/[0-9a-f]+\.qcom,msm-core/uio/uio[0-9]+(/.*)?			u:object_r:sysfs_uio_file:s0
+/sys/devices(/platform)?/soc\.0/[0-9a-f]+\.qcom,dsp_sharedmem/uio/uio[0-9]+(/.*)?			u:object_r:sysfs_uio_file:s0
+/sys/devices(/platform)?/soc\.0/[0-9a-f]+\.qcom,mdm_sharedmem/uio/uio[0-9]+(/.*)?			u:object_r:sysfs_uio_file:s0
+/sys/devices(/platform)?/soc\.0/[0-9a-f]+\.qcom,rmtfs_sharedmem/uio/uio[0-9]+(/.*)?			u:object_r:sysfs_uio_file:s0
+/sys/devices(/platform)?/soc\.0/[0-9a-f]+\.qcom,sensors_sharedmem/uio/uio[0-9]+(/.*)?			u:object_r:sysfs_uio_file:s0
+/sys/devices/[0-9a-f]+\.rmtfs_sharedmem/uio/uio[0-9]+(/.*)?         u:object_r:sysfs_uio_file:s0
+/sys/devices/[0-9a-f]+\.dsp_sharedmem/uio/uio[0-9]+(/.*)?           u:object_r:sysfs_uio_file:s0
+/sys/devices/[0-9a-f]+\.mdm_sharedmem/uio/uio[0-9]+(/.*)?           u:object_r:sysfs_uio_file:s0
+/sys/devices/[0-9a-f]+\.bimc_sharedmem/uio/uio[0-9]+(/.*)?          u:object_r:sysfs_uio_file:s0
+/sys/devices/[0-9a-f]+\.oem_data_sharedmem/uio/uio[0-9]+(/.*)?      u:object_r:sysfs_uio_file:s0
diff --git a/legacy-common/rfs_access.te b/legacy-common/rfs_access.te
index e58a66e..df374b3 100644
--- a/legacy-common/rfs_access.te
+++ b/legacy-common/rfs_access.te
@@ -1,2 +1,4 @@
 allow rfs_access self:socket ioctl;
 allowxperm rfs_access self:socket ioctl msm_sock_ipc_ioctls;
+
+r_dir_file(rfs_access, sysfs_uio_file)
-- 
2.7.4

